sdk/dapp-kit/src/components/connect-modal/ConnectModal.tsx

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import type { WalletWithRequiredFeatures } from '@mysten/wallet-standard';
+import * as Dialog from '@radix-ui/react-dialog';
+import clsx from 'clsx';
+import { useState } from 'react';
+import type { ReactNode } from 'react';
+
+import { useConnectWallet } from '../../hooks/wallet/useConnectWallet.js';
+import { BackIcon } from '../icons/BackIcon.js';
+import { CloseIcon } from '../icons/CloseIcon.js';
+import * as styles from './ConnectModal.css.js';
+import { ConnectionStatus } from './views/ConnectionStatus.js';
+import { GettingStarted } from './views/GettingStarted.js';
+import { WhatIsAWallet } from './views/WhatIsAWallet.js';
+import { WalletList } from './wallet-list/WalletList.js';
+
+type ConnectModalView = 'getting-started' | 'what-is-a-wallet' | 'connection-status';
+
+type ConnectModalProps = {
+	trigger: ReactNode;
+};
+
+export function ConnectModal({ trigger }: ConnectModalProps) {
+	const [isConnectModalOpen, setConnectModalOpen] = useState(false);
+	const [currentView, setCurrentView] = useState<ConnectModalView>();
+	const [selectedWallet, setSelectedWallet] = useState<WalletWithRequiredFeatures>();
+	const { mutate, isError } = useConnectWallet();
+
+	const connectWallet = (wallet: WalletWithRequiredFeatures) => {
+		setCurrentView('connection-status');
+		mutate({ wallet }, { onSuccess: () => setConnectModalOpen(false) });
+	};
+
+	const resetSelection = () => {
+		setSelectedWallet(undefined);
+		setCurrentView(undefined);
+	};
+
+	const onOpenChange = (open: boolean) => {
+		if (!open) {
+			resetSelection();
+		}
+		setConnectModalOpen(open);
+	};
+
+	let modalContent: ReactNode | undefined;
+	switch (currentView) {
+		case 'what-is-a-wallet':
+			modalContent = <WhatIsAWallet />;
+			break;
+		case 'getting-started':
+			modalContent = <GettingStarted />;
+			break;
+		case 'connection-status':
+			modalContent = selectedWallet ? (
+				<ConnectionStatus
+					selectedWallet={selectedWallet}
+					hadConnectionError={isError}
+					onRetryConnection={connectWallet}
+				/>
+			) : null;
+			break;
+		default:
+			modalContent = <WhatIsAWallet />;
```

Comment:
this is a little un-intuitive, but supporting both the mobile and desktop UIs without duplicating a bunch of DOM logic / adding JS is actually kind of tricky.

There is no view selected by default on mobile, so we just say `currentView` is initially undefined. However, we want to display the "What is a wallet" view by default on desktop hence this default case